Sources/Error Handling/ErrorUtils.swift

```diff
@@ -591,3 +561,36 @@ private extension ErrorUtils {
         }
     }
 }
+
+extension Error {
+
+    func addingUserInfo(_ userInfo: [String: Any]) -> Error {
+        let nsError = self as NSError
+        let nsErrorWithUserInfo = NSError(domain: nsError.domain,
+                                          code: nsError.code,
+                                          userInfo: nsError.userInfo + userInfo)
+        return nsErrorWithUserInfo as Error
+    }
+
+}
+
+/// Represents where an `Error` was created
+struct ErrorSource {
+
+    let file: String
+    let function: String
+    let line: UInt
+
+}
+
+/// `Equatable` conformance allows `Error` types that contain source information
+/// to easily conform to `Equatable`.
+extension ErrorSource: Equatable {
+
+    /// However, for ease of testing, we don't actually care if the source of the errors matches
+    /// since expectations will be created in the test and therefore will never match.
+    static func == (lhs: ErrorSource, rhs: ErrorSource) -> Bool {
+        return true
```

Comment:
This feels bad, I understand the idea, but I'm thinking we might as well conform properly, it should be pretty quick and easy.

Reply:
From the docs:
> `Equatable` conformance allows `Error` types that contain source information
> to easily conform to `Equatable`.
> However, for ease of testing, we don't actually care if the source of the errors matches
> since expectations will be created in the test and therefore will never match.

It needs to be said that this is an `internal` type (like our `internal` error types), so it's a hacky implementation detail that won't leak to customers.

Any alternative solutions welcome though :/
It would be equivalent to manually implementing `==` for the error `enum`s (which is very cumbersome) and ignoring the source.